Opt wpdevelopment.ca into our quirk for mouse focusable forms

brentfulgham opened the pull request:
#### 255c3a0f7c54f6a69a6517ff1a6df059288710e7
<pre>
Opt wpdevelopment.ca into our quirk for mouse focusable forms
<a href="https://bugs.webkit.org/show_bug.cgi?id=302673">https://bugs.webkit.org/show_bug.cgi?id=302673</a>
&lt;<a href="https://rdar.apple.com/problem/156109518">rdar://problem/156109518</a>&gt;

Reviewed by Tim Nguyen and Brandon Stewart.

The Wilmot Condos website needs form elements to be mouse focusable to work properly.
This is not the default behavior for Safari, but lacking this behavior breaks their
site preventing users from viewing manuals and other useful information.

Add a quirk to opt this site into the behavior expected by the web developer.

* Source/WebCore/page/Quirks.cpp:
(WebCore::handleWPDevelopmentQuirks):
(WebCore::Quirks::determineRelevantQuirks):

Canonical link: <a href="https://commits.webkit.org/303226@main">https://commits.webkit.org/303226@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f5e7fd791f0c42fc6299bceb39767293839b29f7

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/131648 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/3980 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/42614 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/139157 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/83524 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/aa3180fc-06d4-4053-be2a-13d3c0b6e92f) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/133518 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/4013 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/3821 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/100502 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/68094 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/b2806e70-7eff-4bdf-a75e-446b588198d4) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/134594 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/2955 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/117986 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/81309 "Passed tests") | | ⏳ 🛠 vision-apple 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/2841 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/668 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/82348 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/111563 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/35960 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/141803 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/3724 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/36479 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/108879 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/3772 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/3425 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/109123 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27661 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/2819 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/114166 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/56934 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/3785 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/32633 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/3612 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/67196 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/3954 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/3715 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

Ahmad-S792 commented on Source/WebCore/page/Quirks.cpp:
```diff
@@ -3274,6 +3274,15 @@ static void handleVimeoQuirks(QuirksData& quirksData, const URL& /* quirksURL */
+    if (quirksDomainString != "wpdevelopment.ca"_s) [[unlikely]]
+        return;
+
+    // weebly.com rdar://48003980
```
Shouldn't it be radar matching this PR? Plus for `wpdevelopment.ca`.

brentfulgham replied:
Yes -- bad copy/paste. Thank you for noticing.

nt1m reviewed (approved):
r=me with Ahmad's comment addressed

stwrt reviewed (approved):
Why are form elements not mouse focusable by default?

karlcow reviewed (approved):
Approved by non-reviewer. :) Thanks. 
Another one for the list of form focusable quirk.

brentfulgham committed:
Opt wpdevelopment.ca into our quirk for mouse focusable forms
https://bugs.webkit.org/show_bug.cgi?id=302673
<rdar://problem/156109518>

Reviewed by Tim Nguyen and Brandon Stewart.

The Wilmot Condos website needs form elements to be mouse focusable to work properly.
This is not the default behavior for Safari, but lacking this behavior breaks their
site preventing users from viewing manuals and other useful information.

Add a quirk to opt this site into the behavior expected by the web developer.

* Source/WebCore/page/Quirks.cpp:
(WebCore::handleWPDevelopmentQuirks):
(WebCore::Quirks::determineRelevantQuirks):

Canonical link: https://commits.webkit.org/303226@main
